### ISSUE TITLE
How to use Template literals with dynamic import?

### ISSUE THREAD

**Issue by Popugune** [zeit/next.js#4100]
How to use Template literals with dynamic import?
`
const Component = dynamic(import(`/components/${kindOf}`))
`
In console i have error

`Module build failed: TypeError ... Cannot read property '0' of undefined`

**Comment by timneutkens** [comment]
You can't do truly dynamic imports. Webpack can't predict the outcome and prebundle.

**Comment by herrstucki** [comment]
@timneutkens This is not true. Webpack can handle dynamic imports where part of the path is static. From [the docs](https://webpack.js.org/api/module-methods/#import-):

>Every module that could potentially be requested on an import() call is included. For example, import(`./locale/${language}.json`) will cause every .json file in the ./locale directory to be bundled into the new chunk. At run time, when the variable language has been computed, any file like english.json or german.json will be available for consumption.

This error is specifically a deficiency in `next/babel`, where only static strings are handled but not template literals (which are of a different AST node type). Another unrelated aspect is that all `import()`s are transformed, even those not wrapped in a call to `dynamic()` – which is problematic for non-component dynamic imports.

**Comment by timneutkens** [comment]
Lets re-open the issue then.

**Comment by stramel** [comment]
Would really love to see this get support added.

**Comment by vjpr** [comment]
> This error is specifically a deficiency in next/babel

@herrstucki You mean because of the `handle-import` babel plugin?

**Comment by herrstucki** [comment]
@vjpr yes, this line fails with a template literal: https://github.com/zeit/next.js/blob/4881cd346b8155314c02a6996538fb48d324111d/server/build/babel/plugins/handle-import.js#L57

**Comment by hoang17** [comment]
@herrstucki is there any solution for this problem?

**Comment by morgs32** [comment]
I'd just like to add that this is especially useful when trying to internationalize an app since it allows us to split up imports from `moment` and `react-intl`.

For reference: https://github.com/yahoo/react-intl/issues/777#issuecomment-270134176

**Comment by timneutkens** [comment]
Fixed in #4639
